src/xtgeo/xyz/_xyz_roxapi.py

```diff
@@ -424,6 +424,14 @@ def _roxapi_export_xyz(
         return
 
     is_polygons = isinstance(self, polygons.Polygons)
+    xyz_columns = [self.xname, self.yname, self.zname]
+    if not all(col in df for col in xyz_columns):
```

Comment:
I pref. sets in these cases.

```suggestion
    if not set(xyz_columns).issubset(df.columns)):
```